python/lsst/meas/extensions/gaap/gaap.py

```diff
@@ -0,0 +1,319 @@
+            flagDefs.add(schema.join(middleName, "flag_bigpsf"), ("The Gaussianized PSF is "
+                                                                  "bigger than the aperture"
+                                                                  ))
+        self.flagHandler = FlagHandler.addFields(schema, name, flagDefs)
+        self.EdgeFlagKey = schema.addField(schema.join(name, "flag_edge"), type="Flag",
+                                           doc="Source is too close to the edge")
+
+    @classmethod
+    def getExecutionOrder(cls) -> float:
+        # Docstring inherited.
+        return cls.FLUX_ORDER
+
+    def convolve(self, exposure: afwImage.Exposure, modelPsf: afwDetection.GaussianPsf,
+                 measRecord: lsst.afw.table.SourceRecord) -> afwImage.Exposure:
+        """Convolve the ``exposure`` to make the PSF same as ``modelPsf``.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Original (full) exposure containing all the sources.
+        modelPsf : `lsst.afw.detection.GaussianPsf`
+            Target PSF to which to match.
+        measRecord : `lsst.afw.tabe.SourceRecord`
+            Record for the source to be measured.
+
+        Returns
+        -------
+        convExp : `lsst.afw.image.Exposure`
+            Subexposure containing the source, convolved to the target seeing.
+            The bounding box of the returned image is typically bigger than
+            that of the footprint by the size of the PSF matching kernel
+            minus 1. The exception to this is if the footprint lies too
+            close to the edge of the ``exposure`` and the bounding box is
+            slighly smaller. The flag_edge is set in such cases.
+        """
+        footprint = measRecord.getFootprint()
+        bbox = footprint.getBBox()
+
+        # The kernelSize is guaranteed to be odd, say 2k+1 pixels (default is
+        # 21). The flux inside the footprint is smeared by k pixels on either
+        # side, which is region of interest. The PSF matching sets NO_DATA mask
+        # bit in the outermost k pixels. To account for these nans in the
+        # edges, the subExposure needs to be expanded by another k pixels.
+        # So grow the bounding box initially by 2k, or kernelSize - 1.
+        pixToGrow = self.config.modelPsfMatch.kernel.active.kernelSize - 1
+        bbox.grow(pixToGrow)
+
+        # The bounding box may become too big and go out of bounds for sources
+        # near the edge. Clip the subExposure to the exposure's bounding box.
+        # Set the flag_edge marking that the bbox of the footprint could not
+        # be grown fully but do not set it as a failure.
+        if not exposure.getBBox().contains(bbox):
+            bbox.clip(exposure.getBBox())
+            measRecord.setFlag(self.EdgeFlagKey, True)
+
+        subExposure = exposure[bbox]
+
+        config = self.config.modelPsfMatch
```

Comment:
You can create a copy of a config like:
```
        config = AssembleCoaddConfig()
        # getattr necessary because subtasks do not survive Config.toDict()
        # exclude connections
        configIntersection = {k: getattr(self.config, k)
                              for k, v in self.config.toDict().items()
                              if (k in config.keys() and k != "connections")}
        config.update(**configIntersection)
```

Reply:
Ah, good to know. But I've restructure this part of the code that I don't need to make a copy of a config file anymore.